linkerd/app/gateway/src/gateway.rs

```diff
@@ -86,6 +91,16 @@ where
                     .headers_mut()
                     .append(http::header::FORWARDED, forwarded_header.clone());
 
+                // If we're forwarding HTTP/1 requests, the old `Host` header
+                // was stripped on the peer's outbound proxy. But the request
+                // should have an updated `Host` header now that it's being
+                // routed in the cluster.
+                if let ::http::Version::HTTP_11 | ::http::Version::HTTP_10 = request.version() {
+                    request
+                        .headers_mut()
+                        .insert(::http::header::HOST, host_header.clone());
```

Comment:
nit: elsewhere in this file, we just import from `http::`. is the leading `::` necessary here? if not, i'd prefer to remove it:
```suggestion
                if let http::Version::HTTP_11 | http::Version::HTTP_10 = request.version() {
                    request
                        .headers_mut()
                        .insert(http::header::HOST, host_header.clone());
```